pyscripts/build_docker_image.py

```diff
@@ -10,193 +10,237 @@
 import colorama
 from colorama import Fore
 
-colorama.init(autoreset=True)
-
-default_repo = "(Azure/azure-iot-sdk-BLAH)"
-all_languages = ["c", "csharp", "python", "pythonpreview", "node", "java"]
-
-parser = argparse.ArgumentParser(description="build docker image for testing")
-parser.add_argument(
-    "--language",
-    help="language to build",
-    type=str,
-    required=True,
-    choices=all_languages,
-)
-parser.add_argument("--repo", help="repo with source", type=str, default=default_repo)
-parser.add_argument(
-    "--commit", help="commit to apply (ref or branch)", type=str, default="master"
-)
-parser.add_argument(
-    "--variant", help="Docker image variant (blank for default)", type=str, nargs="?", const=""
-)
-args = parser.parse_args()
-
-if args.repo == default_repo:
-    if args.language == "pythonpreview":
-        args.repo = "Azure/azure-iot-sdk-python-preview"
-    else:
-        args.repo = "Azure/azure-iot-sdk-" + args.language
-    print(Fore.YELLOW + "Repo not specified.  Defaulting to " + args.repo)
-
-print_separator = "".join("/\\" for _ in range(80))
-
-auth_config = {
-    "username": os.environ["IOTHUB_E2E_REPO_USER"],
-    "password": os.environ["IOTHUB_E2E_REPO_PASSWORD"],
-}
-
-
-def get_dockerfile_directory(tags):
-    script_dir = os.path.dirname(os.path.realpath(__file__))
-    return os.path.normpath(os.path.join(script_dir, "../ci-wrappers/" + tags.language))
-
-
-def print_filtered_docker_line(line):
-    try:
-        obj = json.loads(line.decode("utf-8"))
-    except:
-        print(line)
-    else:
-        if "status" in obj:
-            if "id" in obj:
-                if obj["status"] not in [
-                    "Waiting",
-                    "Downloading",
-                    "Verifying Checksum",
-                    "Extracting",
-                    "Preparing",
-                    "Pushing",
-                ]:
-                    print("{}: {}".format(obj["status"], obj["id"]))
+class BuildDockerImage:
```

Comment:
Was there a reason you put this into a class?  Inserting the spaces makes the delta almost impossible to read because (almost) every single line in the file changed.